internal/p2p/pex/reactor.go

```diff
@@ -198,7 +194,11 @@ func (r *Reactor) processPexCh(ctx context.Context) {
 			}
 		// inbound requests for new peers or responses to requests sent by this
 		// reactor
-		case envelope := <-incoming:
+		case envelope, ok := <-incoming:
+			if !ok {
+				r.logger.Debug("incoming channel closed", "ch_id", r.pexCh.ID)
```

Comment:
I think it's not super useful to log for shutdown related things. Either it works (great) or it doesn't (probably deadlocks in which case we sigabrt and get a stack trace, or it panics in which case we get a stack trace,) and in the cases where it doesn't work, we can figure out what happened without logs. 

(Also the debug level is so noisy that I suspect that it's rarely used.